pkg/kademlia/peer_discovery.go

```diff
@@ -15,12 +15,13 @@ import (
 )
 
 type peerDiscovery struct {
-	client node.Client
-	target dht.NodeID
-	opts   discoveryOptions
-
-	cond  sync.Cond
-	queue *XorQueue
+	client   node.Client
+	target   dht.NodeID
+	opts     discoveryOptions
+	foundOne chan *pb.Node
+	foundAll chan []*pb.Node
```

Comment:
It's not sending anything on this channel so it probably shouldn't contain anything. Keep the whitespace, it's necessary to clearly indicate cond protection.

Reply:
This channel part is still WIP cause I was focusing on foundOne first